internal/pkg/saelastic/client.go

```diff
@@ -53,21 +58,61 @@ func (ec *ElasticClient) InitAllMappings() {
 	*/
 }
 
+// createHttpsClient creates http.Client for elastic.Client with enabled
+// cert-based authentication
+func createTlsClient(serverName string, certFile string, keyFile string, caFile string) (*http.Client, error) {
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		log.Fatal(err)
+		return &http.Client{}, err
+	}
+
+	ca, err := ioutil.ReadFile(caFile)
+	if err != nil {
+		log.Fatal(err)
+		return &http.Client{}, err
+	}
+	certPool := x509.NewCertPool()
+	certPool.AppendCertsFromPEM(ca)
+
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		RootCAs:      certPool,
+	}
+	if len(serverName) == 0 {
+		tlsConfig.InsecureSkipVerify = true
+	} else {
+		tlsConfig.ServerName = serverName
+	}
+	tlsConfig.BuildNameToCertificate()
+	return &http.Client{
+		Transport: &http.Transport{TLSClientConfig: tlsConfig},
+	}, nil
+}
+
 //CreateClient   ....
-func CreateClient(elastichost string, resetIndex bool, debug bool) (*ElasticClient, error) {
-	//c, _ = client.New(client.WithHosts([]string{"https://elasticsearch:9200"}))
-	if debug {
+func CreateClient(config saconfig.EventConfiguration) (*ElasticClient, error) {
+	if config.Debug {
 		debuges = func(format string, data ...interface{}) { log.Printf(format, data...) }
 	}
+
 	var elasticClient *ElasticClient
-	//var eClient *elastic.Client
-	eclient, err := elastic.NewClient(elastic.SetHealthcheckInterval(5*time.Second), elastic.SetURL(elastichost))
+	elasticOpts := []elastic.ClientOptionFunc{elastic.SetHealthcheckInterval(5 * time.Second), elastic.SetURL(config.ElasticHostURL)}
+	// add transport with TLS enabled in case it is required
+	if config.UseTls {
+		tlsClient, err := createTlsClient(config.TlsServerName, config.TlsClientCert, config.TlsClientKey, config.TlsCaCert)
```

Comment:
@paramite If `UseTls` is `true` but the `ServerName` was never set, will this fail?

Reply:
Then len(TlsServerName) should be zero. IMO it should work, but can surely add unit test for this case.

Reply:
@paramite I just happened to see this late last night, so I'm not 100% sure if there is an issue here or not. Might be worth just adding the unit test to know for sure.

Are these unit tests currently run in Travis? I assume we need to spin up an ElasticSearch for these to be operational?

Reply:
Yup will add that case ot the suite. Yes, the whole suite runs in Travis. Currently only  connection attempt with TLS is tested, but the success of the connection is not. It will require tweaks to have TLS enabled on ES side in Travis CI job.